### DISCUSSION TITLE
6.0.1 Breaking Changes

### DISCUSSION THREAD

**Discussion by Orgbrat** [Blazored/Modal#289] [Q&A] [answered]
So is there a good guide on the breaking changes of 6.0.1 where you are upgrading from version 5.1.0 Blazor Server?

**Comment by larsk2009** [comment, marked as answer]
There isn't really a list of changes I think. You can always compare between release tags, like [this for example](https://github.com/Blazored/Modal/compare/v5.1.0...v6.0.0). If you look at the files in the `samples` folder, you can see what the only change is:

`Close` => `CloseAsync`
`Cancel` => `CancelAsync`

As you can see in this picture below.
![image](https://user-images.githubusercontent.com/7924568/106658110-27fb8d00-659d-11eb-8690-6be28af7fe85.png)

@chrissainty Maybe we can add a remark to what changes are breaking changes when making a new (major) release?

**Reply by chrissainty** [reply]
@larsk2009 I think that's a great idea. I guess we could do it as release notes in the readme? Or a wiki page perhaps?

**Reply by larsk2009** [reply]
Yeah both are fine with me. Also I think it would be good to put them in the release notes of the release themselves, like [on this page](https://github.com/Blazored/Modal/releases).

**Comment by lwcwew** [comment]
Please change the https://github.com/Blazored/Modal tutorial at step 3 Add CascadingBlazoredModal Component to tell folks the existing Router code in App.razor should be remove then the step 3 code inserted. If folks do not do that then they will get this error: InvalidOperationException: Blazored.Modal.BlazoredModal requires a cascading parameter of type IModalService. Thanks
Microsoft Visual Studio Community 2019 Preview Version 16.9.0 Preview 4.0

**Reply by chrissainty** [reply]
The existing router code shouldn't be removed.

**Reply by larsk2009** [reply]
@chrissainty I think what is going wrong, is that people are not just adding `<CascadingBlazoredModal />` like it says it that step, but they just copy the whole content that is shown in step 3 and *add* it to the existing **App.razor**. If you do that, you end up with 2 routers in **App.razor**, which doesn't work. This was for example the issue of #290.

**Reply by chrissainty** [reply]
@larsk2009 Ahh, OK. Perhaps we can trim the code example so it just shows the opening and closing router elements and just some dots in between? What do you think?

**Reply by larsk2009** [reply]
Yeah seems good to me. To me the existing thing was pretty clear, but I can understand it might be problematic if you are less familiar with Blazor.

**Comment by lwcwew** [comment]
Sorry about my ignorance. Buying your book. If I understand correctly then
you might try this as the example code:
<CascadingBlazoredModal>
--- existing Router code ---
</CascadingBlazoredModal>
anyway thank you for being so kind.



 Jesus washing Peter's feet
Ed Watkins
PO Box 170
Daniel, WY 83115
Phone 760.671.0742
Get in good trouble, necessary trouble,
and help redeem the soul of America.
